core/src/test/java/test/retryAnalyzer/dataprovider/RetryAnalyzerWithDataProviderTest.java

```diff
@@ -26,4 +32,46 @@ public void testRetryCounts() {
     assertEquals(tla.getSkippedTests().size(), 9);
   }
 
+  @DataProvider(name = "getTestData")
+  public Object[][] getTestData() {
+    return new String[][]{
+            new String[]{"abc1", "cdf1"}};
+  }
+
+  @Test(dataProvider = "getTestData", retryAnalyzer = DataProviderRetryAnalyzer.class)
```

Comment:
The split between the test and the test-case is not clear.
Could you move the test-case into a dedicated class `...Sample.java` and execute it from the test?

Reply:
Done.Thanks for your review @krmahadevan  and @juherr!!!!